File: Data Structures & Algorithms/two-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        let container = new Map();
        for (let i = 0; i < nums.length; i++) {
            let diff = target - nums[i];
            if (container.has(diff)) {
                return [container.get(diff), i];
            }
            container.set(nums[i], i)

        }
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        for(int i = 0; i < numbers.length;i++){
            int l = i + 1, r = numbers.length - 1;
            int complement = target - numbers[i];
            while(l <= r){
                int m = l + (r - l) / 2;
                if(numbers[m] < complement){
                    l = m + 1;
                }else if(numbers[m] > complement) {
                    r = m - 1;
                }else {
                    return new int[]{i + 1, m + 1};
                }
            }
        }
        return new int[]{};
    }
}
